### PR TITLE
Use RoleBindings instead of ClusterRoleBindings where possible

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

After discussions with some users, it seems that our current RBAC setup is easy to use but gives us wider rights than many would prefer.

This PR changes some of the `ClusterRoleBindings` to `RoleBindings`. This will make installation for multiple namespaces a bit harder, however it will not give us access for example to secrets in the whole cluster. So it should improve security.

### Checklist

- [x] Make sure all tests pass
- [x] Update documentation
- [x] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

File: documentation/book/con-cluster-operator-rbac.adoc [hunk 1]
```diff
@@ -64,46 +64,66 @@ The Cluster Operator will not run under the cluster admin account.
...
+.`ClusterRole` with namespaced resources for the Cluster Operator
+----
+include::examples/install/cluster-operator/020-ClusterRole-strimzi-cluster-operator-role.yaml[]
+----
+
+The second includes the permissions needed for cluster-scoped resources.
+
 [source,yaml,options="nowrap"]
-.Example `Role` for the Cluster Operator
+.`ClusterRole` with cluster-scoped resoruces for the Cluster Operator
```
Comment (tombentley) [inline documentation/book/con-cluster-operator-rbac.adoc]: typo

File: documentation/book/con-what-the-cluster-operator-does.adoc [hunk 2]
```diff
@@ -15,9 +15,15 @@ On {OpenShiftName} such a cluster can be deployed using the Source2Image feature
 image::cluster_operator.png[Cluster Operator]
 
 When the Cluster Operator is up, it starts to _watch_ for certain {ProductPlatformName} resources containing the desired Kafka or Kafka Connect cluster configuration.
-A `Kafka` resource is used for Kafka cluster configuration, and a `KafkaConnect` resource is used for Kafka Connect cluster configuration.
+By default it watches only in the same namespace or project where it is installed.
+Cluster operator can be configured to watch for more namespaces or projects.
```
Comment (tombentley) [inline documentation/book/con-what-the-cluster-operator-does.adoc]: The Cluster Operator can be...

File: documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+* Modify the installation files according to the namespace the Cluster Operator is going to be installed in.
++
+[source, subs="+quotes"]
+----
+sed -i 's/namespace: .\*/namespace: _<my-namespace>_/' examples/install/cluster-operator/*RoleBinding*.yaml
+----
+
+.Procedure
+
+. Modify the file `examples/install/cluster-operator/050-Deployment-strimzi-cluster-operator.yaml` and in the environment variable `STRIMZI_NAMESPACE` list all namespaces or projects where Cluster Operator should watch for resources.
```
Comment (tombentley) [inline documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc]: I think we need to parameterize `{namespace}`. Also for this to make proper sense it should be something like "Kubernetes namespaces or OpenShift projects" .

File: documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+On {OpenShiftName} this can be done using `oc apply`:
++
+[source,shell,subs=+quotes]
+oc apply -f examples/install/cluster-operator -n _<my-project>_
+
+. Verify whether the Cluster Operator has been deployed successfully, the {KubernetesName} Dashboard or the command line:
++
+[source]
+----
+kubectl describe all
```
Comment (tombentley) [inline documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc]: I've said this before, but this step is meaningless unless you tell the user what to look for in the output.

File: documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc (hunk 3 above)
Comment (scholzj) [reply documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc]: OpenShift is using namespaces. We are using namespaces. We are not using any of the _project_ functionality. I do not think we need to push this too much.

File: documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc (hunk 4 above)
Comment (scholzj) [reply documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc]: I have said it before as well. I have removed it.

File: documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc (hunk 3 above)
Comment (scholzj) [reply documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc]: Ok, I decided to do it at the end. Still not sure this is needed, but whatever.

File: documentation/book/con-cluster-operator-rbac.adoc [hunk 5]
```diff
@@ -64,46 +64,66 @@ The Cluster Operator will not run under the cluster admin account.
 
 The `ClusterRoles` follow the _principle of least privilege_ and contain only those privileges needed by the Cluster Operator to operate Kafka, Kafka Connect, and Zookeeper clusters. The first set of assigned privileges allow the Cluster Operator to manage {ProductPlatformName} resources such as `StatefulSets`, `Deployments`, `Pods`, and `ConfigMaps`.
 
+Cluster Operator is using two different `ClusterRoles`.
```
Comment (dmuntima) [inline documentation/book/con-cluster-operator-rbac.adoc]: change it to: Cluster Operator uses `ClusterRoles` to grant permission at the namespace-scoped resources level and cluster-scoped resources level: 

    ClusterRole with namespace-scoped resources for the Cluster Operator

    link:examples/install/cluster-operator/020-ClusterRole-strimzi-cluster-operator-role.yaml[]

    clusterRole with cluster-scoped resources for the Cluster Operator

    link:examples/install/cluster-operator/021-ClusterRole-strimzi-cluster-operator-role.yaml[]

File: documentation/book/con-what-the-cluster-operator-does.adoc [hunk 6]
```diff
@@ -15,9 +15,15 @@ On {OpenShiftName} such a cluster can be deployed using the Source2Image feature
 image::cluster_operator.png[Cluster Operator]
 
 When the Cluster Operator is up, it starts to _watch_ for certain {ProductPlatformName} resources containing the desired Kafka or Kafka Connect cluster configuration.
-A `Kafka` resource is used for Kafka cluster configuration, and a `KafkaConnect` resource is used for Kafka Connect cluster configuration.
+By default it watches only in the same namespace or project where it is installed.
+The Cluster Operator can be configured to watch for more {Namespaces}.
+The resources Cluster Operator is watching for are:
```
Comment (dmuntima) [inline documentation/book/con-what-the-cluster-operator-does.adoc]: - By default it watches > By default, it watches 

- The resources Cluster Operator is watching for are: > Cluster Operator watches the following resources:

File: documentation/book/con-what-the-cluster-operator-does.adoc [hunk 7]
```diff
@@ -15,9 +15,15 @@ On {OpenShiftName} such a cluster can be deployed using the Source2Image feature
...
 
-When a new desired resource (that is, a `Kafka` or `KafkaConnect` resource) is created in the {ProductPlatformName} cluster, the operator gets the cluster configuration from the desired resource and starts creating a new Kafka or Kafka Connect cluster by creating the necessary other {ProductPlatformName} resources, such as StatefulSets, Services, ConfigMaps, and so on.
+* A `Kafka` resource for Kafka cluster.
```
Comment (dmuntima) [inline documentation/book/con-what-the-cluster-operator-does.adoc]: for Kafka cluster > for the Kafka cluster.
for Kafka Connect cluster. > for the Kafka Connect cluster.
for Kafka Connect cluster. > for the Kafka Connect cluster....

File: documentation/book/con-what-the-cluster-operator-does.adoc [hunk 8]
```diff
@@ -15,9 +15,15 @@ On {OpenShiftName} such a cluster can be deployed using the Source2Image feature
...
+* A `KafkaConnect` resource for Kafka Connect cluster.
+* A `KafkaConnectS2I` resource for Kafka Connect cluster with Source2Image support.
+
+When a new desired resource (that is, a `Kafka`, `KafkaConnect`, or `KafkaConnectS2I` resource) is created in the {ProductPlatformName} cluster, the operator gets the cluster description from the desired resource and starts creating a new Kafka or Kafka Connect cluster by creating the necessary other {ProductPlatformName} resources, such as StatefulSets, Services, ConfigMaps, and so on.
```
Comment (dmuntima) [inline documentation/book/con-what-the-cluster-operator-does.adoc]: When a new desired resource (that is, a `Kafka`, `KafkaConnect`, or `KafkaConnectS2I` resource) is  > When a new `Kafka`, `KafkaConnect`, or `KafkaConnectS2I` resource is created

File: documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc [hunk 9]
```diff
@@ -0,0 +1,68 @@
+// Module included in the following assemblies:
+//
+// assembly-cluster-operator.adoc
+
+[id='deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces{context}']
+= Deploying the Cluster Operator to watch multiple namespaces
+
+.Prerequisites
+
+* Modify the installation files according to the {Namespace} the Cluster Operator is going to be installed in.
```
Comment (dmuntima) [inline documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc]: Modify > Edit . This is to maintain consistency across the guide.

File: documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
++
+[source, subs="+quotes"]
+----
+sed -i 's/namespace: .\*/namespace: _<my-namespace>_/' examples/install/cluster-operator/*RoleBinding*.yaml
+----
+
+.Procedure
+
+. Modify the file `examples/install/cluster-operator/050-Deployment-strimzi-cluster-operator.yaml` and in the environment variable `STRIMZI_NAMESPACE` list all the {Namespaces} where Cluster Operator should watch for resources.
```
Comment (dmuntima) [inline documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc]: Modify > Edit .

File: documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
++
+endif::Kubernetes[]
+On {OpenShiftName} this can be done using `oc apply`:
++
+[source,shell,subs=+quotes]
+oc apply -f examples/install/cluster-operator/020-RoleBinding-strimzi-cluster-operator.yaml -n _<my-project>_
+oc apply -f examples/install/cluster-operator/031-RoleBinding-strimzi-cluster-operator-entity-operator-delegation.yaml -n _<my-project>_
+oc apply -f examples/install/cluster-operator/032-RoleBinding-strimzi-cluster-operator-topic-operator-delegation.yaml -n _<my-project>_
+
+. Deploy the Cluster Operator
```
Comment (dmuntima) [inline documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc]: Deploy the Cluster Operator > Create or update the resource. This is to maintain consistency across the guide.

Comment (scholzj) [reply documentation/book/proc-deploying-cluster-operator-kubernetes-to-watch-multiple-namespaces.adoc]: This should be the same situation as [here](http://strimzi.io/docs/master/full.html#deploying-cluster-operator-openshift-str) or [here](http://strimzi.io/docs/master/full.html#deploying-cluster-operator-kubernetes-str). Deploying the cluster operator here is different from creating or updating the cluster resources. So I think we should keep this.

File: test/src/main/java/io/strimzi/test/TestUtils.java [hunk 12]
```diff
@@ -41,19 +41,19 @@
...
 
-    public static final String CRD_KAFKA_CONNECT_S2I = "../examples/install/cluster-operator/04-Crd-kafkaconnects2i.yaml";
+    public static final String CRD_KAFKA_CONNECT_S2I = "../examples/install/cluster-operator/042-Crd-kafkaconnects2i.yaml";
 
     private TestUtils() {
         // All static methods
     }
 
-    public static final String KAFKA_USER_CRD = "../examples/install/cluster-operator/04-Crd-kafkauser.yaml";
+    public static final String KAFKA_USER_CRD = "../examples/install/cluster-operator/044-Crd-kafkauser.yaml";
```
Comment (ppatierno) [inline test/src/main/java/io/strimzi/test/TestUtils.java]: Just a nit, CRD_KAFKA_USER as the other ones

**Pull request merged**